src/common/procedure/src/error.rs

```diff
@@ -111,4 +118,50 @@ impl Error {
             source: BoxedError::new(err),
         }
     }
+
+    /// Creates a new [Error::RetryLater] error from source `err`.
+    pub fn retry_later<E: ErrorExt + Send + Sync + 'static>(err: E) -> Error {
+        Error::RetryLater {
+            source: BoxedError::new(err),
+        }
+    }
+
+    /// Determine whether it is a retry later type through [StatusCode]
+    pub fn is_retry_later(status_code: StatusCode) -> bool {
```

Comment:
This reminds me that we could add a method like `is_retryable` to the StatusCode